src/Servers/Kestrel/Core/src/Internal/Http/HttpParser.cs

```diff
@@ -446,7 +492,7 @@ private static bool TryTakeSingleHeader(TRequestHandler handler, ReadOnlySpan<by
 
         // Range end is exclusive, so add 1 to valueEnd
         valueEnd++;
-        handler.OnHeader(name: headerLine[..nameEnd], value: headerLine[valueStart..valueEnd]);
+        handler.OnHeader(name: headerLine.Slice(0, nameEnd), value: headerLine[valueStart..valueEnd]);
```

Comment:
Do you have any particular reason for disliking the `headerLine[..nameEnd]` syntax?

Reply:
As mentioned above, there's a codegen quality issue: https://github.com/dotnet/roslyn/issues/43598

Reply:
The range-syntax is nicer to read, so please up-vote the Roslyn-issue to get it fixed asap 😉

Reply:
I see. Good catch @gfoidl! I guess the other places we use the nicer syntax are on the failure paths.